server/authorization.go

```diff
@@ -149,7 +149,7 @@ func (a *AuthorizationServer) Check(ctx gocontext.Context, req *authv3.CheckRequ
 			log.Debugf("no authorization requirements")
 			if err := envRequest.PrepareVariables(); err != nil {
 				log.Errorf("failed to populate context variable: %v", err)
```

Comment:
Let's not log at error level unless there is a fault code. This is related to the ticket I brought up this morning.

Reply:
Just to point out, this change has not been done as part of this or earlier PR related to ARC faults.
Could you explain the reason for removing this logging now ?

Reply:
Sorry this is a wrong line to comment. I meant to point out the log.Errorf in `handleFault()` method

Reply:
I might not have enough context here but could you share more details on why would we not want to log if there is no specific fault code ? You mentioned that service will be overwhelmed but why do we expect errors to be that many in the first place ?
Also, even that log statement did exist for some of the errors before these changes, since I consolidated the different handleX() methods, it is now there for all of them.

Reply:
It's not EA's error if there is a 40x and we usually only log at debug level for those. It's not uncommon for clients to send bad requests and get 404, while we have access log for metrics purpose, the binary itself does not log those 404s as errors. If you look at Envoy's info log, it shows basically nothing related to the traffic it handles.

Could you point out for me what errors were being logged previously that got consolidated here?

Reply:
unavailable() and internalError() were logging. Ref:https://github.com/apigee/apigee-remote-service-envoy/pull/363/files#diff-dc5119ef55fed479882d6318bb13250ddea13884bfdb6aafa3f3359b0e433280L518

So, should we downgrade this to debug log ? or do conditional logging for ARC only ?

Reply:
yeah those two cases are good ones to log errors since it's ARC's problem. The rest should only be debug logs. I believe in those two cases, the fault code will not be empty right? So we just need to conditionally log errors there. The rest can go to debug for troubleshooting purposes.

Reply:
In general, I believe the question is: Is logging XXX useful for support? We should log at >= INFO for cases where that is true.

Reply:
downgraded to log.debug level.